src/metabase/models/permissions.clj

```diff
@@ -1079,6 +1079,49 @@
     (when (and has-native-perms? (not (group-has-native-perms? group-or-id database-or-id)))
       (grant-native-readwrite-permissions! group-or-id database-or-id))))
 
+; Audit Permissions helper fns
+
+(defenterprise default-audit-db-id
+  "OSS implementation of `audit-db/default-audit-db-id`, which is an enterprise feature, so does nothing in the OSS
+  version."
+  metabase-enterprise.audit-db [] ::noop)
+
+(defenterprise default-audit-collection-entity-id
+  "OSS implementation of `audit-db/default-audit-collection-entity-id`, which is an enterprise feature, so does nothing in the OSS
+  version."
+  metabase-enterprise.audit-db [] [])
+
+(defn update-audit-collection-permissions
+  "Will remove or grant audit db (AppDB) permissions, if the instance analytics permissions changes."
+  [group-id changes]
+  (when config/ee-available?
+    (let [audit-collection-id  (-> (t2/select-one 'Collection :entity_id (default-audit-collection-entity-id))
+                                   :id)]
+      (doseq [[change-id type] changes]
+        (when (= change-id audit-collection-id)
+          (let [change-permissions! (case type
+                                      :read  grant-permissions!
+                                      :none  delete-related-permissions!
+                                      :write (throw (ex-info (tru
+                                                              (str "Unable to make audit collections writable."))
+                                                              {:status-code 400})))]
+            (change-permissions! group-id (str "/db/" (default-audit-db-id) "/schema/"))))))))
```

Comment:
```suggestion
            (change-permissions! group-id (all-schemas-path (default-audit-db-id)))))))))
```

Reply:
Although per the [product doc](https://www.notion.so/metabase/Audit-v2-collections-and-navigation-bb26e544848c4dac9a89256d5cac3447?pvs=4#eeeb88f6c6824b28902f166a003ef776) we actually just want to give access to the specific views used by audit queries and keep everything else as `no self service`.

I think you could just hard-code the list of view names, query the DB to find their IDs, and use those to generate the permission paths that we should be writing here.